bgpd/bgp_updgrp_adv.c

```diff
@@ -755,37 +755,39 @@ void subgroup_default_originate(struct update_subgroup *subgrp, int withdraw)
 	}
 
 	if (peer->default_rmap[afi][safi].name) {
+		struct attr attr_tmp = attr;
+		struct bgp_path_info bpi_rmap = {0};
+
+		bpi_rmap.peer = bgp->peer_self;
+		bpi_rmap.attr = &attr_tmp;
+
 		SET_FLAG(bgp->peer_self->rmap_type, PEER_RMAP_TYPE_DEFAULT);
+
+		/* Iterate over the RIB to see if we can announce
+		 * the default route. We announce the default
+		 * route only if route-map has a match.
+		 */
 		for (rn = bgp_table_top(bgp->rib[afi][safi]); rn;
 		     rn = bgp_route_next(rn)) {
 			for (ri = bgp_node_get_bgp_path_info(rn);
 			     ri; ri = ri->next) {
```

Comment:
Not sure if last time I missed. Inner loop to go over path is not required anymore as path is set to self.

Reply:
Actually this is needed in cases when we want to match on specific prefix.

Reply:
:+1:

Reply:
Thanks for fixing this. Latest fix looks good to me.